src/bscPlugin/completions/CompletionsProcessor.ts

```diff
@@ -194,341 +198,156 @@ export class CompletionsProcessor {
-                            label: member.name.text,
-                            kind: isFieldStatement(member) ? CompletionItemKind.Field : CompletionItemKind.Function
-                        });
-                    }
-                }
-            }
-        }
-        return results;
-    }
-
-    public getClassFromMReference(file: BrsFile, position: Position, currentToken: Token, functionScope: FunctionScope): FileLink<ClassStatement> | undefined {
-        let previousToken = file.getPreviousToken(currentToken);
-        if (previousToken?.kind === TokenKind.Dot) {
-            previousToken = file.getPreviousToken(previousToken);
-        }
-        if (previousToken?.kind === TokenKind.Identifier && previousToken?.text.toLowerCase() === 'm' && isMethodStatement(functionScope.func.functionStatement)) {
-            return { item: file.parser.references.classStatements.find((cs) => util.rangeContains(cs.range, position)), file: file };
-        }
-        return undefined;
+    private getSymbolsCompletion(symbols: BscSymbol[], areMembers = false): CompletionItem[] {
+        return symbols.map(symbol => {
+            // if this is low priority, sort it at the end of the list
+            const sortText = symbol.data?.completionPriority ? 'z'.repeat(symbol.data?.completionPriority) + symbol.name : undefined;
+            return {
+                label: symbol.name,
+                kind: this.getCompletionKindFromType(symbol.type, symbol.data, areMembers),
+                detail: symbol?.type?.toString(),
+                documentation: this.getDocumentation(symbol),
+                sortText: sortText
+            };
+        });
     }
 
-    private getGlobalClassStatementCompletions(file: BrsFile, currentToken: Token, parseMode: ParseMode): CompletionItem[] {
-        if (parseMode === ParseMode.BrightScript) {
-            return [];
-        }
-        let results = new Map<string, CompletionItem>();
-        let completionName = file.getPartialVariableName(currentToken, [TokenKind.New])?.toLowerCase();
-        if (completionName?.includes('.')) {
-            return [];
+    private getDocumentation(symbol: BscSymbol) {
+        if (symbol.data?.description) {
+            return symbol.data?.description;
         }
-        let scopes = file.program.getScopesForFile(file);
-        for (let scope of scopes) {
-            let classMap = scope.getClassMap();
-            for (const key of [...classMap.keys()]) {
-                let cs = classMap.get(key).item;
-                if (!results.has(cs.name.text)) {
-                    results.set(cs.name.text, {
-                        label: cs.name.text,
-                        kind: CompletionItemKind.Class
-                    });
-                }
-            }
+        const root = symbol.data?.definingNode?.getRoot();
+        let parser: Parser;
+        if (isBody(root)) {
+            parser = root.parser;
```

Comment:
This is a hack part 3
Basically, I just want to be able to get tokens from an expression statement, so I can look for comments
Is there a better way to do this?